Improve number input field validation

theskumar opened the pull request:
Fixes #3213 
Fixes #3194 

- Allow the user to enter any input
- Display instant feedback if the input is not a valid number.

It uses alpine and `isNaN` javascript function for validation
Also, move alpine to be included in both public and apply site; forms right now doesn't use `base_apply.html`

Checkout the video below for it's look and feel.

https://user-images.githubusercontent.com/236356/222098169-a0a57d65-eb8f-4851-9b11-29f1860317c6.mp4

frjo commented on hypha/templates/base-apply.html:
```diff
@@ -42,11 +42,6 @@
-        <script defer src="{% static 'js/apply/vendor/alpine-3.11.1.min.js' %}"></script>
-        <style> [x-cloak] {display: none !important} </style>
-        <!-- apline.js ends-->
-
```
This should not be removed I think?

theskumar replied:
You're right! not sure why I thought `base_apply.html` was inherited from the `base.html`.

theskumar replied:
Updated.

frjo reviewed (changes requested):
Alipinejs got removed in the base-apply template.

theskumar committed:
Improve number input field validation

- Allow the user to enter any input
- Display instant feedback if the input is not a valid number.

It uses alpine and isNaN javascript functions for validation

Fixes: #3213 #3194
